### DISCUSSION TITLE
Does PRQL support SQL FILTER clauses?

### DISCUSSION THREAD

**Discussion by jcuenod** [PRQL/prql#4928] [Q&A]
Filter is overloaded here, so to be clear, I am referring to this kind of filter clause: https://www.sqlite.org/windowfunctions.html#the_filter_clause

In the sqlite docs, they have this example:

```sql
SELECT c, a, b, group_concat(b, '.') FILTER (WHERE c!='two') OVER (
  ORDER BY a
) AS group_concat
FROM t1 ORDER BY a;
```

This is a bit convoluted because it's also part of a window function. A simpler example is:

```sql
SELECT
  year,
  group_concat(title, ', ') FILTER (WHERE genre = 'comedy') as comedies,
  group_concat(title, ', ') FILTER (WHERE genre = 'drama') as dramas
FROM movies
GROUP BY year;
```

Is there PRQL that would compile to that SQL?

**Comment by max-sixty** [comment]
No, unfortunately not at the moment. 

TBH this is the first I've heard of it. Is this exclusively a sqlite feature? It seems similar in some ways to `QUALIFY`...

**Reply by jcuenod** [reply]
I've used it in [Clickhouse](https://clickhouse.com/docs/knowledgebase/filtered-aggregates) and [DuckDB](https://duckdb.org/docs/sql/query_syntax/filter.html) before. I see that it's also available in [Postgres](https://www.postgresql.org/docs/current/sql-expressions.html#SYNTAX-AGGREGATES).

See https://modern-sql.com/feature/filter

**Reply by max-sixty** [reply]
Ah great, that is nice. Particularly in a simple aggregate like DuckDB allows.

I think this could be great to include in PRQL in some way. There's a very old issue which seems like it could be reassessed:  https://github.com/PRQL/prql/issues/82 

We'd need to think of the PRQL semantics — we would really want it to be consistent with the existing structure. Possibly the examples in that issue could work; would need to think more.

**Reply by jcuenod** [reply]
Yes, that's exactly the kind of thing I'm wondering about (I think the `case ... when` syntax is actually just an alternative for the same thing)!
